File: cart-service/src/main/java/com/example/cart_service/service/CartService.java
package com.example.cart_service.service;

import com.example.cart_service.dto.response.CartResponse;

public interface CartService {

    CartResponse addToCart(Long userId, Long productId, Integer quantity);

    CartResponse getCart(Long userId);

    CartResponse updateItem(Long userId, Long itemId, Integer quantity);

    CartResponse removeItem(Long userId, Long itemId);

    void clearCart(Long userId);
}

File: sam-shop-frontend/src/app/features/cart/cart.service.ts
import { Injectable } from '@angular/core';
import { Cart } from '../../core/models/cart.model';
import { ApiService } from '../../core/services/api.service';
import { TokenService } from '../../core/services/token.service';

@Injectable({ providedIn: 'root' })
export class CartService {
  constructor(private api: ApiService, private tokenService: TokenService) {}

  add(productId: number, quantity: number) {
    return this.api.post<Cart>('/api/cart/add', { productId, quantity });
  }

  getCart() {
    const userId = this.tokenService.getUser()?.userId;
    return this.api.get<Cart>(`/api/cart/${userId}`);
  }

  update(itemId: number, quantity: number) {
    return this.api.put<Cart>('/api/cart/update', { itemId, quantity });
  }

  remove(itemId: number) {
    return this.api.delete<Cart>(`/api/cart/remove/${itemId}`);
  }
}
